src/main/java/com/kodilla/ecommercee/api/controllers/CartController.java

```diff
@@ -1,42 +1,46 @@
 package com.kodilla.ecommercee.api.controllers;
 
-import com.kodilla.ecommercee.domain.ProductDto;
+import com.kodilla.ecommercee.domain.CartDto;
+import com.kodilla.ecommercee.exceptions.CartNotFoundException;
+import com.kodilla.ecommercee.services.CartService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @RequestMapping("/api/v1/cart")
+@RequiredArgsConstructor
 public class CartController {
 
-    private List<ProductDto> cart;
+    private final CartService cartService;
 
-    @GetMapping(value = "/cart/")
-    public ResponseEntity<Void> createCart() {
-        cart = new ArrayList<>();
-        return ResponseEntity.ok().build();
+    @GetMapping(value = "{cartId}")
+    public ResponseEntity<CartDto> getCart(@PathVariable Long cartId) throws CartNotFoundException {
+        return ResponseEntity.ok(cartService.getCart(cartId));
     }
 
     @GetMapping
-    public ResponseEntity<List<ProductDto>> getProducts() {
-        return ResponseEntity.ok(new ArrayList<>());
+    public ResponseEntity<List<CartDto>> getCarts() {
+        return ResponseEntity.ok(cartService.getAllCarts());
     }
 
     @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Void> addProduct(@RequestBody ProductDto productDto) {
+    public ResponseEntity<Void> createCart(@RequestBody CartDto cartDto) {
+        cartService.saveCart(cartDto);
         return ResponseEntity.ok().build();
     }
 
-    @DeleteMapping(value = "{productId}")
-    public ResponseEntity<Void> deleteProduct(@PathVariable Long productId) {
+    @DeleteMapping(value = "{cartId}")
+    public ResponseEntity<Void> deleteCart(@PathVariable Long cartId) throws CartNotFoundException {
+        cartService.deleteCart(cartId);
         return ResponseEntity.ok().build();
     }
 
-    @PostMapping(value = "/order/")
-    public ResponseEntity<Void> createOrder(@RequestBody List<ProductDto> cart) {
-        return ResponseEntity.ok().build();
+    @PutMapping
+    public ResponseEntity<CartDto> updateCart(@RequestBody CartDto cartDto) {
```

Comment:
Oczekiwane jest ResponseEntity<Void> createOrder(@RequestBody List<ProductDto> cart) - tworzymy zamówienie na podstawie produktów. Tu jest pole do dyskusji jak to zrobić

Reply:
Mam pomysł, żeby w momencie utworzenia zamówienia była opcja w kontrolerze wykasowania wszystkich pozycji z karty i metoda createOrder by to robiła, natomiast pobranie wartości dokonywałoby się przed tym już za pomocą kontrolera OrderController